overlord/snapstate/snapmgr.go

```diff
@@ -42,33 +41,31 @@ type SnapManager struct {
 	runner *state.TaskRunner
 }
 
-type installState struct {
-	Name    string              `json:"name"`
-	Channel string              `json:"channel"`
-	Flags   snappy.InstallFlags `json:"flags,omitempty"`
+type snapSetup struct {
+	Name    string `json:"name"`
+	Version string `json:"version"`
+	Channel string `json:"channel"`
 
-	DownloadTaskID string `json:"download-task-id,omitempty"`
-	SnapPath       string `json:"snap-path,omitempty"`
-}
+	OldName    string `json:"old-name"`
+	OldVersion string `json:"old-version"`
 
-type downloadState struct {
-	Developer string `json:"developer"`
-	SnapPath  string `json:"snap-path,omitempty"`
-}
+	InstallFlags snappy.InstallFlags `json:"inst-flags,omitempty"`
+	RemoveFlags  snappy.RemoveFlags  `json:"rm-flags,omitempty"`
 
-type removeState struct {
-	Name  string             `json:"name"`
-	Flags snappy.RemoveFlags `json:"flags,omitempty"`
+	SnapPath string `json:"snap-path"`
 }
 
-type rollbackState struct {
-	Name    string `json:"name"`
-	Version string `json:"version,omitempty"`
+func (s *snapSetup) BaseDir() string {
```

Comment:
interestingly Info now is going to have a BaseDir method too, still wondering where we'll need to make Infos out of setup data